crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -0,0 +1,130 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind};
+use biome_console::markup;
+
+use biome_js_syntax::JsStringLiteralExpression;
+
+use biome_rowan::AstNode;
+use regex::Regex;
+use std::sync::LazyLock;
+
+// List of sensitive patterns
+const SENSITIVE_PATTERNS: &[&str] = &[
```

Comment:
These strings are eventually made regex, so there's no value in creating a static string and generating a `Regex` **at runtime** inside a loop, that's why you have that huge regression in performance.

Instead, create the `Regex` as a `const`, at compile time. Those regex aren't dynamic, and don't depend from the user input. And that's where you can LazyLock too.

I won't block the PR, but we should fix the performance regression before merging the PR

Reply:
Good point @ematipico. I will refactor this according to several comments with useful input towards a robust and DX-friendly solution.

Reply:
Based on this, I've also added some multithreading. I don't know how acceptable this according to biome's code practices, as I found it mostly in underlying libraries and not in the rules themselves. However, it seems like usecases like this one might require some multithreading.